src/sanitizers.rs

```diff
@@ -133,6 +133,49 @@ pub fn strip_stuff_between_messages(s: &str) -> String {
         .collect()
 }
 
+/// Remove excess lines on tag 86 statements beyond the 6 allowed.
+///
+/// Note that you potentionally lose information with this sanitizer
+pub fn strip_excess_tag86_lines(input: &str) -> String {
+    let mut lines_to_delete = vec![];
+
+    let tag_86_lines = input.lines().enumerate().filter_map(|(line, contents)| {
+        if contents.starts_with(":86") {
+            Some(line)
+        } else {
+            None
+        }
+    });
+
+    for line_no in tag_86_lines {
+        let lines = input.lines().skip(line_no + 1);
+
+        // find all lines excess of the 5 allowed additional lines
+        let to_delete = lines
+            .enumerate()
+            .take_while(|(_, contents)| !contents.starts_with(':'))
+            .filter_map(move |(line, _)| {
+                if line >= 5 {
+                    Some(line + line_no + 1)
+                } else {
+                    None
+                }
+            });
+
+        lines_to_delete.extend(to_delete);
+    }
+
+    let filtered_lines = input.lines().enumerate().filter_map(|(line, contents)| {
+        if !lines_to_delete.contains(&line) {
+            Some(format!("{}\r\n", contents))
```

Comment:
This copies every single line, every time, even if it's not overlong since `format!()` will always allocate a `String`. Can you find a more efficient solution? It might not matter after all but thankfully we have benchmarks. After you add `strip_excess_tag86_lines()` to `sanitize()`, you can check the benchmark suite and see whether we take a hit. If we don't, then might as well leave the current implementation be.

Reply:
I think ideally you'd use `.intersperse("\r\n").collect()` here, but `iter.itersperse()` is currently nightly-only. So what seems to be at least slightly better is using `.collect::<Vec<&str>>().join("\r\n")`.